File: AppSets-iOS/AppSets/com/app/appsets/im/MessageBroker.swift
//
//  MessageBroker.swift
//  AppSets
//
//  Created by caiju Xu on 2024/7/2.
//

import Foundation

protocol MessageBroker {
    
    associatedtype C:MessageBrokerConfig
    
    func bootstrap(_ config: C)
    
    func retry()
    
    func close()
    
    func sendMessage(_ imObj:ImObj, _ imImMessage:any ImMessage) async -> Bool
}

File: AppSets-Android/appsets/src/main/kotlin/xcj/app/appsets/im/MessageBroker.kt
package xcj.app.appsets.im

import xcj.app.appsets.im.message.IMMessage

interface MessageBroker<C : MessageBrokerConfig> {

    suspend fun bootstrap(config: C)

    suspend fun retry()

    suspend fun close()

    suspend fun sendMessage(imObj: IMObj, imMessage: IMMessage)

}
